homeassistant/components/notify/pushetta.py

```diff
@@ -18,45 +18,59 @@
 
 
 CONF_CHANNEL_NAME = 'channel_name'
+CONF_SEND_TEST_MSG = 'send_test_msg'
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
     vol.Required(CONF_API_KEY): cv.string,
     vol.Required(CONF_CHANNEL_NAME): cv.string,
+    vol.Optional(CONF_SEND_TEST_MSG, default=False): cv.boolean,
 })
 
 
 def get_service(hass, config):
     """Get the Pushetta notification service."""
-    from pushetta import Pushetta, exceptions
+    pushetta_service = PushettaNotificationService(config[CONF_API_KEY],
+                                                   config[CONF_CHANNEL_NAME],
+                                                   config[CONF_SEND_TEST_MSG])
 
-    try:
-        pushetta = Pushetta(config[CONF_API_KEY])
-        pushetta.pushMessage(config[CONF_CHANNEL_NAME],
-                             "Home Assistant started")
-    except exceptions.TokenValidationError:
-        _LOGGER.error("Please check your access token")
-        return None
-    except exceptions.ChannelNotFoundError:
-        _LOGGER.error("Channel '%s' not found", config[CONF_CHANNEL_NAME])
-        return None
-
-    return PushettaNotificationService(config[CONF_API_KEY],
-                                       config[CONF_CHANNEL_NAME])
+    if pushetta_service.is_valid:
+        return pushetta_service
 
 
 # pylint: disable=too-few-public-methods
 class PushettaNotificationService(BaseNotificationService):
     """Implement the notification service for Pushetta."""
 
-    def __init__(self, api_key, channel_name):
+    def __init__(self, api_key, channel_name, send_test_msg):
         """Initialize the service."""
-        from pushetta import Pushetta
+        from pushetta import Pushetta, exceptions
         self._api_key = api_key
         self._channel_name = channel_name
-        self.pushetta = Pushetta(self._api_key)
+        self.is_valid = True
+
+        try:
+            self.pushetta = Pushetta(api_key)
+        except exceptions.TokenValidationError:
```

Comment:
When I wrote the platform this didn't work and was the reason for sending the message. There is no exception if you create the object with the wrong API key but when you want to send the message. No need for this try...except block.

```python
>>> from pushetta import Pushetta, exceptions
>>> pushetta = Pushetta('test-api-key')
>>> pushetta.pushMessage('test', 'message')
Traceback (most recent call last):
....
    raise TokenValidationError("Invalid token")
pushetta.exceptions.TokenValidationError: Invalid token
```